hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/SqmCorrelatedRootJoin.java

```diff
@@ -55,7 +49,7 @@ public SqmCorrelatedRootJoin(
 
 	@Override
 	public SqmRoot<T> getCorrelationParent() {
-		return null;
+		return this;
```

Comment:
Is there a failing test due to how it was before? There are other classes following the same approach, that's why I am asking

Reply:
the failing test id `OuterJoinTest#testJoinOrderWithRightJoinWithInnerImplicitJoins`